### PR TITLE
adding edit page to appeals

### PR DESCRIPTION
**Pull request by jiahuang**
Connects #7594 

appeal_edit1

To do:
- [x] add tests

### CODE REVIEW

File: app/controllers/appeals_controller.rb [hunk 1]
```diff
@@ -67,8 +67,36 @@ def show
     end
   end
 
+  helper_method :appeal, :url_appeal_uuid
+
+  def appeal
+    @appeal ||= Appeal.find_appeal_by_id_or_find_or_create_legacy_appeal_by_vacols_id(params[:id])
```
Comment (shanear) [inline app/controllers/appeals_controller.rb]: lol `find_appeal_by_id_or_find_or_create_legacy_appeal_by_vacols_id` is a hell of a method name

File: app/models/appeal.rb [hunk 2]
```diff
@@ -32,6 +32,13 @@ def self.find_appeal_by_id_or_find_or_create_legacy_appeal_by_vacols_id(id)
     end
   end
 
+  def ui_hash(ama_enabled)
+    super.merge(
+      docketType: docket_type,
```
Comment (shanear) [inline app/models/appeal.rb]: shouldn't these be snake_case per our discussion :)

File: spec/feature/intake/edit_spec.rb [hunk 3]
```diff
@@ -43,6 +45,54 @@ def check_row(label, text)
...
+      # going back to edit page should show those issues
+      visit "appeals/#{appeal.uuid}/edit/"
+      expect(page).to have_content("Left knee granted")
+      expect(page).not_to have_content("nonrating description")
+
+      # canceling should redirect to queue
+      click_on "Cancel edit"
+      expect(page).to have_current_path("/queue/appeals/#{appeal.uuid}")
+    end
+  end
```
Comment (shanear) [inline spec/feature/intake/edit_spec.rb]: nice test. Love the comments

File: app/models/request_issues_update.rb [hunk 4]
```diff
@@ -38,14 +38,18 @@ def perform!
   def process_end_product_establishments!
     attempted!
 
-    review.process_end_product_establishments!
+    # appeals do not have process_end_product_establishments
+    if review.respond_to?(:process_end_product_establishments!)
```
Comment (shanear) [inline app/models/request_issues_update.rb]: I'm thinking we shouldn't even do this async stuff for appeals. Can we just do `attempted!` and `processed!` synchronously for appeals and guard this whole method on `return unless review.respond_to?(:process_end_product_establishments!)`?

File: app/models/decision_review_intake.rb [hunk 5]
```diff
@@ -19,4 +19,21 @@ def cancel_detail!
   def review_errors
     detail.errors.messages
   end
+
+  def complete!(request_params)
```
Comment (shanear) [inline app/models/decision_review_intake.rb]: I don't think it's super clear what conceptually belongs in the yield block here. Maybe take a block through the `&block` way  (see: https://mixandgo.com/learn/mastering-ruby-blocks-in-less-than-5-minutes) and name it something more descriptive? (like `&inside_transaction`) Or add a comment?

File: app/models/appeal.rb (hunk 2 above)
Comment (leikkisa) [reply app/models/appeal.rb]: 👍

**Pull request merged**